lib/Driver/Driver.cpp

```diff
@@ -2031,6 +2031,12 @@ const ToolChain *Driver::getToolChain(const ArgList &Args) const {
       TC = new toolchains::Darwin(*this, Target);
       break;
     case llvm::Triple::Linux:
+      if (Target.isAndroid()) {
```

Comment:
Totally agree. One potential wart is the fact that other parts of the codebase will reference things like `os(Windows)`, whereas the toolchain will be named `Cygwin`. Even with that in mind, I still prefer naming the toolchain `Cygwin`, though.